csrc/multidevice/utils.cpp

```diff
@@ -228,7 +228,7 @@ IterDomain* getShardedIterDomain(
     NVF_THROW("Unexpected parallel type: ", parallel_type);
   }();
 
-  for (auto&& [index, id] : enumerate(domain)) {
+  for (auto&& [index, id] : enumerate(domain | TensorDomain::kNoReductions)) {
```

Comment:
```suggestion
  for (IterDomain* id : domain | TensorDomain::kNoReductions) {
```
`index` doesn't seem to be used